### ISSUE TITLE
Batch_size in Empty Latent Image results in all generated images to have the same starting seed in the metadata (wrong seed)

### ISSUE THREAD

**Issue by Yasand123** [comfyanonymous/ComfyUI#3130]
If `batch_size` is more than 1 in the `Empty Latent Image` node, all the generated images will have the starting seed in the metadata.

For example, the current seed is 1000. You change the batch_size to 4 to generate 4 images at a time. All of these 4 images will have the same seed of 1000 in their metadata. Which is not correct. Only the first generated image accurately has that seed.

This is an issue because I currently have multiple images that were generated that way and I will never know what seed they used because of this. In my humble opinion, I think this is an important issue to address.

**Comment by woferry** [comment]
Just adding a +1 here.  I understand the workflow being preserved and the seed value that was used being kept, but seems like the "real seed" for that particular image should still be available somewhere in the saved data, or else perhaps indicating which run of the batch that particular image was if that would always result in a consistent seed value (though if it's random that wouldn't work).  As a workaround I saw it mentioned that one could use the XY Plot feature and Seed++ since it at least writes the seed values on the final Plot image, though that only increments it doesn't do random seeds (I guess I'll see if I can create my own custom node to fix that).

**Comment by ltdrdata** [comment]
There is a need to accurately understand the concept of seed.
A seed is the initial value used to generate noise.

When generating images with 4 batches in A1111, it creates noise for each of the 4 separate latents by incrementally increasing the seed and applying it to each.

When generating images with 4 batches in ComfyUI, it's not treated as 4 separate latents, but as one combined latent of 4, and the noise is applied in a single step. This means that the image generated from the second latent cannot be reproduced simply by using the starting seed + 1, but only by using the starting seed + the second latent.

There is no unique seed for the second latent.

To reprodcue, you need to use `Latent From Batch` to select, as shown in the following workflow.
However, this method cannot be applied to stochastic samplers such as `_sde` or `_ancestral`. 
This is because they include randomness that is not controlled by the seed.

![workflow (22)](https://github.com/user-attachments/assets/316b9d89-3a5e-4b74-935b-63713543f06a)

To summarize, in ComfyUI, there is no such thing as unique metadata for a single image. When multiple outputs are generated from one workflow, there exists a single workflow as metadata for all of them.

**Comment by woferry** [comment]
In that case, it seems like knowing the index in the batch number would totally work, then.  I.e. if it was the 4th image in a batch of >4, there would just need to be a step to "throw away" the first 3 chunks of the latent generated, and use the 4th one to re-create the image, right?
